cmake/leatherman.cmake

```diff
@@ -156,6 +160,94 @@ macro(enable_cpplint)
     endif()
 endmacro()
 
+# Usage: gettext_templates(dir)
+#
+# Create templates for gettext in `dir` from the source files found in additional specified dirs
+# Creates a custom target `translation`.
+macro(gettext_templates dir)
+    find_program(XGETTEXT_EXE xgettext)
+
+    if (XGETTEXT_EXE)
+        set(TRANSLATION_DIR "${dir}")
+        file(MAKE_DIRECTORY ${TRANSLATION_DIR})
+
+        file(GLOB_RECURSE ALL_PROJECT_SOURCES ${ARGN})
```

Comment:
This is a duplicate GLOB_RECURSE with the one we do at the top level for cpplint. Does it make sense to just do that glob earlier in the toplevel CMakeLists.txt so wqe can pass the same list of files there and here, instead of globbing?

All of our other projects should have a similar glob for lint, so I don't think that this is undue work for other repos